app/components/common/navbar/index.test.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { unstable_createRemixStub as createRemixStub } from '@remix-run/testing';
+import { render } from '@testing-library/react';
+import Slider from '../slider/index';
+import Navbar, { DynamicHeroIcon } from './index';
+
+describe('Navbar', () => {
+  // Make the stub
+  const RemixStub = createRemixStub([
+    {
+      path: '/',
+      element: <Navbar />,
+    },
+  ]);
+
+  it('renders', () => {
+    const { container } = render(<RemixStub />);
+    expect(container).toBeInTheDocument();
+  });
+
+  it('renders the slider', () => {
+    const props = {
+      showSlider: true,
+      setShowSlider: jest.fn(),
+    };
+    const { container } = render(<Slider isOpen={props.showSlider} toggle={props.setShowSlider} />);
+    expect(container).toBeInTheDocument();
+  });
+
+  it('renders the dynamic hero icon', () => {
+    const props = {
+      name: 'random',
+      className: 'random',
+    };
+    const { container } = render(<DynamicHeroIcon name={props.name} className={props.className} />);
+
+    expect(container).toBeInTheDocument();
+  });
+
+  // simulate click on li
+  // it('should simulate click on li', async () => {
+  //   const navbarElements = {
+  //     navbarPages: [
+  //       {
+  //         icon: 'Home',
+  //         text: 'Home',
+  //         href: '/',
+  //         id: 0,
+  //         visibleOnDesktop: true,
+  //       },
+  //       {
+  //         icon: 'Calendar',
```

Comment:
Are the comments required here @isVivek99

Reply:
we need to fix this test, once we get a fix simulating react-router's working in nodejs environment. That is why i was proposing to move to `vitest`. eg: [here](https://github.com/mcansh/stubbbbbs/blob/main/test/setup-test-env.ts)